Add Terraform provider docs content

nrichers opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

## Description
<!--- Describe your changes in detail. -->

This PR adds content for the Terraform provider docs on elastic.co.

## Related Issues
<!--- This project only accepts pull requests related to open issues. -->
<!--- If suggesting a new feature or change, please discuss it in an -->
<!--- issue first.  If fixing a bug, there should be an issue describing -->
<!--- it with steps to reproduce.  Please link to the any related issues -->
<!--- here: -->

Relates to 
* https://github.com/elastic/terraform-provider-ec/issues/145
* https://github.com/elastic/docs/pull/1995

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes.  Include -->
<!--- details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

Built locally, looks like this: 

![image](https://user-images.githubusercontent.com/15148011/101686292-b6301880-3a1d-11eb-9f8d-8ebf5e3b5553.png)

~(Note the double horizontal lines. Need to look into what we need to get rid of one of these.)~ Fixed this with Greg's help.

## Types of Changes
<!--- What types of changes does your code introduce? Put an `x` in -->
<!--- all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation

## Readiness Checklist
<!--- Go over all the following points, and put an `x` in all the boxes -->
<!--- that apply.  If you're unsure about any of these, don't hesitate -->
<!--- to ask. We're here to help! -->
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

karencfv reviewed (commented):
Thanks for the PR @nrichers! I have only one concern. As the `docs/` directory is what is visible on the terraform registry, I am wondering the implications of having a second file named `index` and a file with the `asciidoc` extension. 

As far as I know we can't ask anyone from the Hashicorp side about this until the legal stuff for the partnership is done. Is it possible for this file to live in a new `docs-internal/` directory or something similar?
I am not entirely sure

karencfv commented on docs/index.asciidoc:
```diff
@@ -0,0 +1,11 @@
+
+The provider lets you manage Elastic Cloud deployments as code, and introduce DevOps-driven methodologies to manage and deploy the Elastic Stack and solutions.
+
+To get started, see our link:https://registry.terraform.io/providers/elastic/ec/latest/docs[Elastic Cloud Terraform provider documentation].
```
This link is not live yet, would it break the docs build?

nrichers replied:
External links are not checked by our elastic.co docs build, so this should be fine.

gtback replied:
☝️

alaudazzi commented:
@nrichers can we check with @gtback if the doc build accepts file types other than Asciidoc?

Kushmaro reviewed (approved):
text LGTM

gtback commented:
> @nrichers can we check with @gtback if the doc build accepts file types other than Asciidoc?

We currently only support Asciidoc. I suppose we could hack in support for copying raw HTML files (we have some minimal support for overriding the "index.html" page for a book right now), but I imagine that gets us further from our goal than closer (maintaining Markdown and HTML). Perhaps we could try automating Pandoc conversion of Markdown -> Asciidoc, as some other projects do? Alternatively, if Hashicorp will host what becomes the "primary" version of the docs, perhaps we should just have a small overview and a link to Hashicorp's site, like we do with some of the Elasticsearch client library pages:
* [Go](https://www.elastic.co/guide/en/elasticsearch/client/go-api/current/overview.html)
* [Ruby](https://www.elastic.co/guide/en/elasticsearch/client/ruby-api/current/ruby_client.html)
* [Rust](https://www.elastic.co/guide/en/elasticsearch/client/rust-api/current/overview.html)

nrichers commented:
> As far as I know we can't ask anyone from the Hashicorp side about this until the legal stuff for the partnership is done. Is it possible for this file to live in a new `docs-internal/` directory or something similar?

@karencfv moving the files is not a problem, I've updated the PRs. I moved the content to `docs-elastic/`, as the internal bit might throw off some people. Happy to move again if need be.

nrichers commented:
> > @nrichers can we check with @gtback if the doc build accepts file types other than Asciidoc?
> 
> We currently only support Asciidoc. I suppose we could hack in support for copying raw HTML files (we have some minimal support for overriding the "index.html" page for a book right now), but I imagine that gets us further from our goal than closer (maintaining Markdown and HTML). Perhaps we could try automating Pandoc conversion of Markdown -> Asciidoc, as some other projects do?

We do automatically convert Markdown to Asciidoc for ecctl command reference content, but it would require additional steps to adapt this as the primary content source here. It may not be worth the effort, given that we have a short timeline and we'll be moving to a new docs system soon.

(Reading between the lines, I do wonder if @alaudazzi wasn't thinking about our forthcoming use of MDX or about using Markdown as an early experiment. She's out this week, but perhaps could comment further when she is back.)

> Alternatively, if Hashicorp will host what becomes the "primary" version of the docs, perhaps we should just have a small overview and a link to Hashicorp's site, like we do with some of the Elasticsearch client library pages

@gtback that's exactly what we are doing with this PR. I think that's likely the best approach for now, given that we use it elsewhere (I borrowed from the Site Search docs, which do something very similar.)

gtback commented:
> @gtback that's exactly what we are doing with this PR. I think that's likely the best approach for now, given that we use it elsewhere (I borrowed from the Site Search docs, which do something very similar.)

Perfect! I wasn't sure if this was just the starting point and if you were planning to expand to the rest of the Markdown files in `/docs` (or if that's what @alaudazzi was asking about).

karencfv reviewed (approved):
Thanks everyone! @nrichers the `docs-elastic` directory is fine with me :)

alaudazzi commented:
Agreed. Thank you @gtback for your feedback. As @nrichers stated in his last comment above, this is the best approach for now.

alaudazzi reviewed (approved):
LGTM.

nrichers commented:
> LGTM.

Why are you approving PRs on your week off, @alaudazzi?! 😁

alaudazzi commented:
@nrichers it was a special week off.

nrichers commented:
Please don't merge this, yet. There are some additional horizontal rules that we need to get rid of. It looks like other books did this by way of injecting some HTML, but I'm still looking into what's required.

gtback commented:
I marked this as a draft so it doesn't accidentally get merged prematurely.

marclop commented:
@nrichers we're going forward with the `v0.1.0-beta` version name, I've commented in elastic/docs#1995 to update it with the `v0.1.0-beta` branch name.

nrichers commented:
OK, @gtback helped me solve the issue of the extra horizontal lines (fixed via an update to https://github.com/elastic/docs/pull/1995). This PR is good to merge.

nrichers commented:
> @nrichers we're going forward with the `v0.1.0-beta` version name, I've commented in [elastic/docs#1995](https://github.com/elastic/docs/pull/1995) to update it with the `v0.1.0-beta` branch name.

@marclop I've created the `v0.1.0-beta` branch off master and will make sure we can build this as part of [elastic/docs#1995].
